zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -470,10 +481,50 @@ def test_901_pause_resume(self):
         with self.pause_resume(['nova-compute']):
             logging.info("Testing pause resume")
 
+    def test_904_test_ceph_keys(self):
+        """Run pause and resume tests.
+
+        Pause service and check services are stopped then resume and check
+        they are started
+        """
+        # Expected default and alternate values
+        if zaza.model.get_application_config(
+                self.application_name)['libvirt-image-backend'].get(
+                    'value') != 'rbd':
+            return
+
+        regex = re.compile(r"^\[client.(.+)\]\n\tkey = (.+)$")
+        key_dict = {}
+
+        def check_keyring(app_name_dict_list):
+            for app_name_dict in app_name_dict_list:
+                for app_name, key_list in app_name_dict.items():
+                    for unit in zaza.model.get_units(
+                            app_name, model_name=self.model_name):
+                        for key_app_name in key_list:
+                            keyring_file = (
+                                '/etc/ceph/ceph.client.{}.keyring'.format(
+                                    key_app_name))
+                            data = str(generic_utils.get_file_contents(
+                                unit, keyring_file))
+
+                            result = regex.findall(data)
+                            self.assertEqual(2, len(result))
+                            self.assertEqual(result[0], key_app_name)
+                            for k, v in key_dict.items():
+                                if k == result[0]:
+                                    self.assertEqual(v, result[1])
+                                else:
+                                    self.assertNotEqual(v, result[1])
+                            key_dict[result[0]] = result[1]
+
+        check_keyring([{self.application_name: [self.application_name]}])
```

Comment:
this was coded for stable/yoga where it does not have the partial fix that is on master. This needs to be adjust to run against the code on master.

Reply:
I rewrote that logic in the new version I am going to push soon